### ISSUE TITLE
Improve Foursquare provider

### ISSUE THREAD

**Issue by AlexArchive** [TerribleDev/OwinOAuthProviders#72]
In #69  @rjperes added the Foursquare provider. It works well, but I would like to see some small improvements made before I publish. 

**Xml Documentation**
- `FoursquareAuthenticationMiddleware` `CreateHandler` method should have documentation. See [here](https://github.com/owin-middleware/OwinOAuthProviders/blob/master/Owin.Security.Providers/GitHub/GitHubAuthenticationMiddleware.cs#L63) for an example.
- `FoursquareAuthenticatedContext` and it's members should have documentation.
- `FoursquareAuthenticationProvider` and it's members should have documentation.
- `FoursquareReturnEndpointContext` should have documentation.
- `IFoursquareAuthenticationProvider` and it's members should have documentation. See this [folder](https://github.com/owin-middleware/OwinOAuthProviders/tree/master/Owin.Security.Providers/GitHub/Provider) for examples.

**Remove Scope**

Foursquare does not support scope. It is misleading to have a `Scope` property on the `FoursquareAuthenticationOptions` class. It should be removed. So should any referencing statements or expressions ([like this one](https://github.com/owin-middleware/OwinOAuthProviders/blob/3906d0147366744638613606e341b2df755c3e59/Owin.Security.Providers/Foursquare/FoursquareAuthenticationHandler.cs#L178)).

**Superfluous hash**

When the provider redirects to the `RedirectUri` - a peculiar hash is appended to the location. 

![](http://i.imgur.com/2FBx3vg.png)

I am not sure what the significance of `_=_` is. It would be nice to remove it.

**Improve the `FoursquareAuthenticatedContext` model**
- Properties who hold Json should be of type `JObject` not `String`.

(I will think about how to improve the model some more - I have a couple of other "feelings" that I cannot very well articulate at the moment)

**Comment by rjperes** [comment]
I added comments to the documentation and removed the scope. Still couldn't remove _=_, not sure where it comes from, but found this http://stackoverflow.com/questions/7131909/facebook-callback-appends-to-return-url.

**Comment by AlexArchive** [comment]
Awesome. 

Could you please submit a pull request and we can take it from there.

**Comment by AlexArchive** [comment]
Hm. I was reading the documentation - researching the superfluous fragment  - and I noticed that the [version parameter](https://developer.foursquare.com/overview/versioning) probably _shouldn't_ be the current date. 

What we have now is somewhat of a ticking time bomb - if Foursquare update the Api and introduce a breaking change - our provider _might_ break.

**Comment by rjperes** [comment]
You're right... will update that... what date shall I choose? Say, January 1st, 2015? Will include that in the pull request.

**Comment by AlexArchive** [comment]
I would go with March 19 personally.

**Comment by rjperes** [comment]
Right.

**Comment by AlexArchive** [comment]
I just sent an e-mail to @foursquare inquiring about the superfluous URL fragment - I did some investigating and it comes from their API. 

![](http://i.imgur.com/JjrWtrb.png)

Hopefully we'll hear back from them soon!

**Comment by rjperes** [comment]
;-)
